providers/openfeature-flagd-provider/lib/openfeature/flagd/provider/client.rb

```diff
@@ -104,7 +108,14 @@ def prepare_evaluation_context(evaluation_context)
         end
 
         def error_response(error_code, error_message)
-          ResolutionDetails.new(error_code, error_message, "ERROR", nil, nil)
+          OpenFeature::SDK::Provider::ResolutionDetails.new(
+            value: nil,
+            reason: "ERROR",
+            variant: nil,
+            error_code: error_code,
+            error_message: error_message,
+            flag_metadata: nil
+          )
```

Comment:
We should discourage providers from returning an error in the resolution details. Instead, the provider should throw and let the SDK create the response. The reason for this is we need the SDK to run the error hooks if something goes wrong in the provider.

Reply:
Ok, is it ok with you if I rework that part in https://github.com/open-feature/ruby-sdk-contrib/pull/25?

Reply:
Sure 👍